pkgs/applications/window-managers/ragnarwm/default.nix

```
@@ -0,0 +1,58 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, writeText
+, fontconfig
+, libX11
+, libXft
+, libXcursor
+, libXcomposite
+, conf ? null
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ragnarwm";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "cococry";
+    repo = "Ragnar";
+    rev = finalAttrs.version;
+    hash = "sha256-SZWhmFNmS2oLdO9BnPzimoind1452v/EEQzadc5A+bI";
+  };
+
+  buildInputs = [
+    fontconfig
+    libX11
+    libXft
+    libXcursor
+    libXcomposite
+  ];
+
+  prePatch = ''
+    mkdir -p $out/bin
+    mkdir -p $out/share/applications
+
+    substituteInPlace Makefile \
+      --replace '/usr/bin' "$out/bin" \
+      --replace '/usr/share' "$out/share"
+  '';
+
+  postPatch =
+  let
+    configFile =
+      if lib.isDerivation conf || builtins.isPath conf
+      then conf else writeText "config.h" conf;
+  in
+  lib.optionalString (conf != null) "cp ${configFile} config.h";
```

`nixpkgs-fmt` says this should be 1 indention level deeper.
```suggestion
  postPatch =
    let
      configFile =
        if lib.isDerivation conf || builtins.isPath conf
        then conf else writeText "config.h" conf;
    in
    lib.optionalString (conf != null) "cp ${configFile} config.h";
```

That's the way it is in `dwm` as well:
https://github.com/NixOS/nixpkgs/blob/8353344d3236d3fda429bb471c1ee008857d3b7c/pkgs/applications/window-managers/dwm/default.nix#L22-L28